primeqa/pipelines/components/reader/generative.py

```diff
@@ -24,3 +32,118 @@ def load(self, *args, **kwargs):
 
     def apply(self, input_texts: List[str], context: List[List[str]], *args, **kwargs):
         pass
+    
+@dataclass
+class GenerativeFiDReader(ReaderComponent):
+    
+    model: str = field(
+        default="PrimeQA/eli5-fid-bart-large-with-colbert-passages",
+        metadata={"name": "Model"},
+    )
+    max_seq_len: int = field(
+        default=256,
+        metadata={
+            "name": "Maximum sequence length",
+            "description": "Maximum length of question and context inputs to the model (in word pieces/bpes)",
+            "range": [32, 512, 8],
+        },
+    )
+    max_answer_length: int = field(
+        default=256, metadata={"name": "Maximum answer length", "range": [2, 2000, 2]}
+    )
+    generation_max_length: int = field(
+        default=256, metadata={"name": "Maximum answer length", "range": [2, 2000, 2]}
+    )
+    generation_num_beams: int = field(
+        default=1, metadata={"name": "The number of beams for generation", "range": [1, 5, 1]}
+    )
+    num_contexts: int = field(
+        default=3, metadata={"name": "The number of passages in the input", "range": [1, 10, 1]}
+    )
+
+    def __post_init__(self):
+        # Placeholder variables
+        self._preprocessor = None
+        self._trainer = None
+    
+    def load(self, *args, **kwargs):
+        task_heads = FID_HEAD
+        # Load configuration for model
+        config = AutoConfig.from_pretrained(self.model)
+
+        # Initialize tokenizer
+        tokenizer = AutoTokenizer.from_pretrained(
+            self.model,
+            config=config,
+        )
+
+        config.sep_token_id = tokenizer.convert_tokens_to_ids(tokenizer.sep_token)
+        model = FiDModelForDownstreamTasks.from_config(
+            config,
+            self.model,
+            task_heads=task_heads,
+        )
+        model.set_task_head(next(iter(task_heads)))
+
+        # Initialize preprocessor
+        self._preprocessor = ELI5FiDPreprocessor(
+            stride=0,
+            max_seq_len=self.max_seq_len,
+            tokenizer=tokenizer,
+            max_contexts=3,# self.num_contexts,
+            max_answer_len=self.generation_max_length
+        )
+
+        data_collator = FiDDataCollator(tokenizer)
+        
+         # Initialize post processor
+        postprocessor = ELI5FiDPostProcessor(
```

Comment:
I guess this is ok for now, but what if you wanted to use a new preprocessor? There is no ability to have a different one right now.